packages/next/src/server/base-server.ts

```diff
@@ -1334,6 +1342,14 @@ export default abstract class Server<ServerOptions extends Options = Options> {
         isSSG =
           isSSG || !!manifest.routes[pathname === '/index' ? '/' : pathname]
       }
+
+      if (isSSG && isFetchAction) {
+        appendPrerenderCookies(res, {
+          previewModeId: opts.previewProps.previewModeId,
+        })
```

Comment:
Is this enabling preview mode on the response from the server action as well?

Ideally we only append the preview cookie in the server fetch request itself to allow invoking the underlying function and not have the preview cookie persist in the session.